vowpalwabbit/stagewise_poly.cc

```diff
@@ -76,9 +76,11 @@ struct stagewise_poly
 #endif  // DEBUG
 
     //synth_ec.feature_space[tree_atomics].delete_v();
-    synth_ec.indices.delete_v();
     free(sd);
     free(depthsbits);
+
+    // Intentionally do not clear the unions here.
+    synth_ec.delete_unions(nullptr, nullptr);
```

Comment:
Does this call actually do anything or is it only here to make it clear that nothing needs to be done?

Reply:
So originally i had debug assertions verifying that for every example the unions were deleted, or at least considered through a call like this. However, due to an issue in memory_tree it's not immediately feasible. I left this in like you said to make it clear that it is not necessary.